### ISSUE TITLE
Allow "inside-building quests"?

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#558]
Currently quests requiring to go into an object are not allowed. However, some users suggested that they might be useful, anyway.

This rule is, however, not always strictly enforced, as these quests may need looking/going inside a building:
* wheelchair quest (I admit, I also sometimes did not do this and just looked whether the entry is okay…) https://github.com/westnordost/StreetComplete/issues/26
* changing table (might not be advertised on the outside) https://github.com/westnordost/StreetComplete/issues/344
* toilets (may not always be visible from the outside?) https://github.com/westnordost/StreetComplete/pull/270
 
Generally this might be a requirement for these quests:
* smoking? https://github.com/westnordost/StreetComplete/issues/539
* internet access? https://github.com/westnordost/StreetComplete/issues/99
* takeaway? https://github.com/westnordost/StreetComplete/issues/104 (here, however, maybe you would even have to ask, as no sign is even inside the building)
* website and contact info? https://github.com/westnordost/StreetComplete/issues/92, https://github.com/westnordost/StreetComplete/issues/454
* there was also some discussion about vegetarian/vegan quest (it may not be visible from the outside) https://github.com/westnordost/StreetComplete/issues/479
* beer? https://github.com/westnordost/StreetComplete/issues/82
* worship service times? (here was some discussion too) https://github.com/westnordost/StreetComplete/issues/232
* maybe more…

(Note this list should not imply all these quests should be disabled by default or restricted with a requirement below.)

Here are some…

# ways to handle them

## disable them by default
As @Etua proposed in https://github.com/westnordost/StreetComplete/issues/554#issuecomment-325229126:
> However as the number of suggestions of quests that require entering some place (or specifically building) will increase (even already implemented wheelchair access may be considered as one) we may think of including them together with #35. Ideally all such tasks wouldn't be shown by default, but a possibility to turn them on as a group or only some of them depending on a user's decision would be offered.

Depends on https://github.com/westnordost/StreetComplete/issues/35

I (@rugk) agreed in https://github.com/westnordost/StreetComplete/issues/553#issuecomment-325345056:
> The best thing would be if #35 is solved and then such quests could be added as "disabled by default".

## fork?
@CloCkWeRX suggested in https://github.com/westnordost/StreetComplete/issues/553#issuecomment-325238302:
> I'd be really keen to explore a gentle fork of StreetComplete focused on "I'm sitting down/standing around in line inside of a business, what can I answer" for things like this, #104 #99 etc.
> You could even call it... "EatComplete" :P

I (@rugk) opposed in https://github.com/westnordost/StreetComplete/issues/553#issuecomment-325345056:
> IMHO a fork would not be nice, (don't wanna use two apps…) better improve this app.

## use location detection
@ServusWorld suggested in https://github.com/westnordost/StreetComplete/issues/553#issuecomment-325304387:
> Maybe such POI-specific quests shouldn't be shown on the map always, but only if you are very near (distance < 15m) to it so that it can be assumed you are inside.

Of course, ideas can be combined. 

# A note about users and gamification

A problem coming with the (maybe in the future even increased) gamification is that users may **answer quests without actually seriously checking what the quests asks for** and thus enter unreliable data.
When they cannot enter a building but see the quest they may be tempted to do that. (This effect is, however, of course also possible without gamification, i.e. just because the user wants to solve as many quests as possible or clear the map or so…)

I see, however, a difference between the **vegan/vegtarian quest** and **wheelchair quest** here e.g. As I said, I incorrectly mapped the wheelchair quest some times or at least, did not check whether the inner part of the building is also accessible. That may have to do with the thing, that the quest talks about whether the building is *access*ible, i.e. whether you can enter there. (Note the German translation literally translates this as "zugänglich".) So such things may happen in a **good faith**.

I however doubt that many users would wrongly answer the vegan/vegetarian quest when they cannot see it from the outside, as here they cannot just guess or so… It is very obvious that both things can be possible and users have no clue when there is no sign with the menu or so…
That's why they'd have to select one option arbitrarily, whcih would be very clear to users that it is vandalism. So here they would have to act in **bad faith**.

I am just telling this, because that may be a difference for these quests and whether they might be shown/disabled/off-by-default or how any other restriction mechanism may be applied.

**Comment by westnordost** [comment]
I expect that when/before I introduce more gamification in the app, we need to put (even) more thought into streamlining quests to be straighforward to answer and eliminating incentives to prematurely solving quests.
So, yeah, "inside-building" quests are such a candidates, because they require relatively much effort for solving just one quest. So, as long as there are still other "outside" quest suggestions around, waiting to be implemented, this has no priority.

If implemented, @ServusWorld 's suggestion makes the most sense, though a little tricky to implement. Also, since many smaller businesses or businesses which do not cover the whole building (i.e. are just in the ground floor of a multi-storey building) are usually mapped as nodes, it could then happen that these "hidden" quests pop up when just standing in front. It will be tricky to really detect that you are *inside* and not just walking by on a normal survey. If these quests are just shown after i.e. one stays at/near the restaurant node for more than 5 minutes, then on the other hand, these quests will be shown and solved very slowly, decreasing their value for implementation effort.

**tl;dr**: Very low priority.

**Comment by rugk** [comment]
>  It will be tricky to really detect that you are inside and not just walking by on a normal survey

Also, usually you may have bad GPS connection *inside* a building and wifi location data or so may be not there or inaccurate.

**Comment by rugk** [comment]
So as https://github.com/westnordost/StreetComplete/pull/672 (filtering of quests and disabled by default state) has been implemented, I think this issue is… well… kinda fixed, is not it?

The current approach is just to disable them and warn the user of the consequences when enabling them. Do we still need other detection things proposed in the OP?

If you are okay with that, I'd go through the list in the OP and comment on the issues for quest suggestions, which may now be implemented as "disabled by default" quests.

**Comment by westnordost** [comment]
Wow, your initial description is quite well written and thought through. I will read through this topic again.

**Comment by westnordost** [comment]
Yeah, okay, I'd consider this as fixed. An automatic detection has its own problems which will open another can of worms, so the current "disabled by default, warning message when activating" approach is the best I think.
